### DISCUSSION TITLE
Support for Multi-Page Applications and Static Exports

### DISCUSSION THREAD

**Discussion by manjime83** [pocketbase/pocketbase#5317] [Q&A]
I would like to open a discussion about the potential for PocketBase to support multi-page applications (MPAs) or to implement functionality that support the static exports feature found in meta frameworks like Next.js, specifically using the [output: 'export'](https://nextjs.org/docs/pages/building-your-application/deploying/static-exports) option, that allows the creation of just static content. It would be a great feature to support such exports, using the pb_bublic folder.

Currently, PocketBase excels in serving single-page applications, but as the landscape of web development evolves, many developers are leaning towards multi-page architectures or static site generation for various reasons, including performance and SEO benefits. While I appreciate the flexibility offered by the [routerUse](https://pocketbase.io/docs/js-routing#middlewares) middleware, I believe there is an opportunity to enhance PocketBase's capabilities by providing a reference implementation that can seamlessly handle static exports from frameworks like Next.js.

The idea would be to enable developers to take the output of a Next.js build and serve that static content directly through PocketBase. This would not only improve the versatility of PocketBase but also attract a wider range of developers who are looking for a robust backend solution that can easily integrate with popular frontend frameworks.

Implementing this feature could involve creating a new middleware or extending the existing routerUse functionality to facilitate the serving of static files.

I would love to hear your thoughts on this proposal. Do you think this aligns with PocketBase’s goals? Are there any existing workarounds or third-party solutions that you think could inform this discussion?

**Comment by ganigeorgiev** [comment]
> ... It would be a great feature to support such exports, using the pb_bublic folder.
> The idea would be to enable developers to take the output of a Next.js build and serve that static content directly through PocketBase.

It should be already supported. What is the error you've encountered?

**Reply by ganigeorgiev** [reply]
It is not Next.js but we embed and render the static export of SvelteKit for the [project website](https://github.com/pocketbase/site) in https://pocketbase.io. If for some reason your Next.js export placed in `pb_public` doesn't work make sure that all exported pages are actually static and doesn't rely on a running node server.

**Reply by manjime83** [reply]
The main issue I had was related to the paths and URLs on the pages. On some paths, when I refreshed the page, I got 404 errors. After investigating further, I found that it was something fixable using the trailing slash in the URLs.
